src/business/entities/movie.ts

```diff
@@ -0,0 +1,67 @@
+export class Movie {
+    constructor(
+        private id: string,
+        private title: string,
+        private date: Date,
+        private length: string,
+        private synopsis: string,
+        private link: string,
+        private picture: string
+        ){ }
+        
+        public getId(): string {
+            return this.id;
+        }
+        
+        public setId(id: string): void {
+            this.id = id;
+        }
+        
+        public getTitle(): string {
+            return this.title;
+        }
+        
+        public setTitle(title: string): void {
+            this.title = title;
+        }
+        
+        public getDate(): Date {
+            return this.date;
+        }
+        
+        public setDate(date: Date): void {
+            this.date = date;
+        }
+        
+        public getLength(): string {
+            return this.length;
+        }
+        
+        public setLength(length: string): void {
+            this.length = length;
+        }
+        
+        public getSynopsis(): string {
+            return this.synopsis;
+        }
+        
+        public setSynopsis(synopsis: string): void {
+            this.synopsis = synopsis;
+        }
+        
+        public getLink(): string | undefined {
```

Comment:
Porque o link pode ser `undefined` se a propriedade `this.link` não foi declarada como opcional? `private link?: string` não está assim!